extensions/github/src/extension.ts

```diff
@@ -0,0 +1,152 @@
+    this._onDidChangeSession.event;
+
+  protected sessions: GithubSession[] = [];
+
+  constructor(private telemetryLogger: extensionApi.TelemetryLogger) {}
+
+  async getSessions(scopes?: string[]): Promise<readonly extensionApi.AuthenticationSession[]> {
+    if (scopes?.length > 0) throw new Error('Scopes are not supported.');
+    return this.sessions;
+  }
+
+  async createSession(scopes: string[]): Promise<extensionApi.AuthenticationSession> {
+    if (scopes.length > 0) throw new Error('Scopes are not supported.');
+
+    // We only support one session
+    if (this.sessions.length) return this.sessions[0];
+
+    // Get the access token from the preferences
+    const accessToken = getGitHubAccessToken();
+
+    // Check if the token exists
+    if (accessToken === undefined || accessToken.length === 0) {
+      await window.showErrorMessage('No access token provided.');
+      throw new Error('No access token provided.');
+    }
+
+    // Validate the token
+    // (1) Create an Octokit client with the access token
+    // (2) Get the /user endpoint to get user information (username email etc.)
+    const octokit = new Octokit({
+      auth: accessToken,
+    });
+    const { data } = await octokit.users.getAuthenticated();
+
+    // Create a session with our valid access token
+    const session = new GithubSession(
+      `${data.id}`,
+      accessToken,
+      {
+        id: `${data.id}`,
+        label: `${data.name}`,
+      },
+      [],
+    );
+
+    this.telemetryLogger.logUsage('github-create-auth-session');
+
+    // Fire events
+    this._onDidChangeSession.fire({
+      added: [session],
+    });
+
+    // Store the session
+    this.sessions.push(session);
+
+    return session;
+  }
+  async removeSession(sessionId: string): Promise<void> {
+    const removed = this.sessions.filter(session => session.id === sessionId);
+    this._onDidChangeSession.fire({
+      removed: removed,
+    });
+    this.sessions = this.sessions.filter(session => session.id !== sessionId);
+  }
+}
+
+export async function activate(extensionContext: extensionApi.ExtensionContext): Promise<void> {
+  const telemetryLogger = extensionApi.env.createTelemetryLogger();
+  const authProvider = new GitHubProvider(telemetryLogger);
+
+  // Register the AuthenticationProvider
+  const disposable = extensionApi.authentication.registerAuthenticationProvider('github', 'GitHub', authProvider, {
+    images: {
+      icon: {
+        light: './logo-light.png',
+        dark: './logo-dark.png',
+      },
+      logo: {
+        light: './logo-light.png',
+        dark: './logo-dark.png',
+      },
+    },
+  });
+  extensionContext.subscriptions.push(disposable);
+  // If we have a defined GitHub preference access token, let's create the session
+  if (getGitHubAccessToken()) {
+    try {
+      await authProvider.createSession([]);
+    } catch (e) {
+      await window.showErrorMessage(`GitHub access token is not valid.`);
+      console.error(e);
+    }
```

Comment:
it looks like we're not testing this part.
Should the error message also be displayed to the user ?

Reply:
It is tested in https://github.com/containers/podman-desktop/blob/2539606fc9a13e476fa208648d4b838e5fe640f4/extensions/github/src/extension.spec.ts#L100